src/Neo/Plugins/Plugin.cs

```diff
@@ -247,33 +247,46 @@ protected internal virtual void OnSystemLoaded(NeoSystem system)
         /// <returns><see langword="true"/> if the <paramref name="message"/> is handled by a plugin; otherwise, <see langword="false"/>.</returns>
         public static bool SendMessage(object message)
         {
+            foreach (var plugin in Plugins)
+            {
+                if (plugin.IsStopped)
+                {
+                    continue;
+                }
 
-            return Plugins.Any(plugin =>
+                bool result;
+                try
                 {
-                    try
-                    {
-                        return !plugin.IsStopped &&
-                               plugin.OnMessage(message);
-                    }
-                    catch (Exception ex)
+                    result = plugin.OnMessage(message);
+                }
+                catch (Exception ex)
+                {
+                    Utility.Log(nameof(Plugin), LogLevel.Error, ex);
+
+                    switch (plugin.ExceptionPolicy)
                     {
-                        switch (plugin.ExceptionPolicy)
-                        {
-                            case UnhandledExceptionPolicy.StopNode:
-                                throw;
-                            case UnhandledExceptionPolicy.StopPlugin:
-                                plugin.IsStopped = true;
-                                break;
-                            case UnhandledExceptionPolicy.Ignore:
-                                break;
-                            default:
-                                throw new InvalidCastException($"The exception policy {plugin.ExceptionPolicy} is not valid.");
-                        }
-                        Utility.Log(nameof(Plugin), LogLevel.Error, ex);
-                        return false;
+                        case UnhandledExceptionPolicy.StopNode:
+                            throw;
```

Comment:
```suggestion
                            throw ex;
```

Reply:
there is no need to add ex. throw will do it already.

Reply:
add it or change `catch (Exception ex)` to just `catch`, no need to allocate memory for a object your going to use.

Reply:
log is using it.

Reply:
And these level of optimization meas nothing to the core, should not put your attention to them.

Reply:
I am, just never ending game of PR waiting. The soon #3408, #3409 and #3410 merged. I can continue. I could realistically put in 50 PR in a week. But what good is that when everyone gone all the time.

Reply:
You can put 50 prs, but wont be merged without being reviewed. And people will get tired if you have many prs that focus on different things. Cause reviewing also take times. To be honest, reviewing pr sometimes may take longer than implementing it.